app/controllers/api/v1/games/shots_controller.rb

```diff
@@ -2,24 +2,28 @@ module Api
   module V1
     module Games
       class ShotsController < ApiController
+        before_action :require_correct_player
+
         def create
-          game = Game.find(params[:game_id])
-          # player = game.send(game.current_turn)
-          # binding.pry
-          # opponent = game.send(game.send(current_turn!))
-          if game.current_turn.last.to_i.odd?
-            player = game.player_1
-            opponent = game.player_2
+          # game = Game.find(params[:game_id])
+
+          if current_game.current_turn.last.to_i.odd?
+            player = current_game.player_1
+            opponent = current_game.player_2
           else
-            player = game.player_2
-            opponent = game.player_1
+            player = current_game.player_2
+            opponent = current_game.player_1
           end
-          # binding.pry
 
-          turn_processor = TurnProcessor.new(game, params[:shot][:target], player, opponent)
+          turn_processor = TurnProcessor.new(current_game, params[:shot][:target], player, opponent)
 
           turn_processor.run!
-          render json: game, message: turn_processor.message
+          render json: current_game, message: turn_processor.message
+        end
+
+        def require_correct_player
+          require 'pry'; binding.pry
+          render status: 400, json: current_game, message: "Invalid move. It's your opponent's turn" unless current_player?
         end
```

Comment:
Just changing these to match our new current_game and current_player API base methods.